### PR TITLE
Feature/51613 - add trap focus in modules Modal and OffCanvasSidebar

### PR DESCRIPTION
**Pull request by niemaco**

### CODE REVIEW

File: components/03-modules/modal/Modal.js [hunk 1]
```diff
@@ -69,36 +67,55 @@ export default {
       this.ariaHidden = state ? 'false' : 'true'
       this.visibility = state
 
-      if (state === true) {
-        this.$nextTick(() => this.$refs.modal.focus())
-      } else {
+      if(state) {
         this.$nextTick(() => this.focused.focus())
+      } else {
+        this.$nextTick(() => this.trigger.focus())
```
Comment (szafran89) [inline components/03-modules/modal/Modal.js]: please check if we can get rid of this condition

File: components/03-modules/modal/Modal.js [hunk 2]
```diff
@@ -69,36 +67,55 @@ export default {
       this.ariaHidden = state ? 'false' : 'true'
       this.visibility = state
 
-      if (state === true) {
-        this.$nextTick(() => this.$refs.modal.focus())
-      } else {
+      if(state) {
```
Comment (szafran89) [inline components/03-modules/modal/Modal.js]: missing space after `if`

Comment (niemaco) [reply components/03-modules/modal/Modal.js]: I removed the above code.

File: components/03-modules/modal/Modal.js (hunk 1 above)
Comment (niemaco) [reply components/03-modules/modal/Modal.js]: I removed unnecessary data "focused" and above statement. I've moved the appropriate focus setting to show and hide methods.

**Pull request merged**